### ISSUE TITLE
Timeouts not clearing in jsPsych.similarity

### ISSUE THREAD

**Issue by rivasdaniel** [jspsych/jsPsych#62]
Hi! I am working with your library to develop an experiment investigating the effects of category learning on similarity judgments at Pr. Stevan Harnad's lab at UQAM in Montreal. Our design requires that trials be limited to a fixed time limit, a feature I am currently implementing, and in the course of development I noticed the following bug in the similarity plugin:

Because the calls to setTimeout() are not cancelled when the anonymous function on line 201 is called, many functions try to be executed after a trial has ended. Calls on line 75, 84, and 105 are still scheduled, and thus happen at innapropriate times during the next trial. This causes erratic behavior. Notice that the sim_trial_complete flag isnt really useful as a fix for this, since it only protects from the call to line 105 during the inter trial period, and if, like me, you have none, then the flag is set to true by the next trial immediately.

In short, the code assumes that the subjects will only submit their answer after having seen both stimuli, and that timing_post_trial > timing_second_stim. As a fix, I propose to set handles to each call to setTimeout, and have the anonymous function that starts the next trial run cleartimeout() on all of them.

By the way, I would love to contribute to this open source project, as I think web experiments are very useful to increase participant pool and mitigate costs. If you are interested, I can also show you how I implemented my timeout feature.

Regards,
Daniel Rivas
